transaction/src/builder/manifest_builder.rs

```diff
@@ -303,13 +301,13 @@ impl ManifestBuilder {
     pub fn take_non_fungibles_from_worktop(
         self,
         resource_address: impl ResolvableResourceAddress,
-        ids: BTreeSet<NonFungibleLocalId>,
+        ids: &BTreeSet<NonFungibleLocalId>,
```

Comment:
I purposefully changed it to remove & to make the interface nicer / so that we could change to an IntoIterator model :) - the caller can just call clone if they need to (very rarely)

Reply:
I wanted to use `AsRef<BTreeSet<NonFungibleLocalId>>` so both reference and owned can be used. We can improve this later.

Reply:
Hmm but then it requires a btreeset, rather than, say, allowing an array. I think I'd prefer `IntoIterator` and then doing a count on the length of each to verify uniqueness :).

Reply:
Ah, I think my main concern was about passing ownership of the non-fungible local ids.

How about `AsRef<[NonFungibleLocalId]>`?

Reply:
I'm not sure I follow the concern over ownership? `BTreeSet<NonFungibleLocalId>` doesn't implement `AsRef<[NonFungibleLocalId]>` I believe